core/src/main/java/com/linecorp/armeria/client/ClientDecorationBuilder.java

```diff
@@ -48,7 +50,19 @@ public ClientDecorationBuilder add(ClientDecoration clientDecoration) {
      * @param decorator the {@link Function} that transforms an {@link HttpClient} to another
      */
     public ClientDecorationBuilder add(Function<? super HttpClient, ? extends HttpClient> decorator) {
-        decorators.add(requireNonNull(decorator, "decorator"));
+        return add(decorator, DecoratorAndOrder.DEFAULT_ORDER);
+    }
+
+    /**
+     * Adds the specified HTTP-level {@code decorator}.
+     *
+     * @param decorator the {@link Function} that transforms an {@link HttpClient} to another
+     * @param order the order of {@code decorator}
+     */
+    public ClientDecorationBuilder add(Function<? super HttpClient, ? extends HttpClient> decorator,
```

Comment:
I know this would have been better in the issue a while ago, but just to think about, instead of explicit order what if it's just e.g. `prependDecorator` or `appendDecorator` based on server or client. Is there a real use case for explicit order, given usually if you have such fine-grained order you can just call the methods in the right order? I'm guessing the main usage is to add to the beginning of the list after the fact. And in that case it's a bit tricky having to look up the default order value to know whether you need to use a negative or positive value to achieve it.

Reply:
When I opened #3137 issue, the first inserted RoutingDecorator has a higher priority, unlike other decorators.
I thought it is natural for a `RoutingDecorator` to ack like a filter. However, it was my mistake #3160 and fixed by #3166.

Now, for all decorators API, the later inserted decorator has a higher priority always. 
Had chat with @minwoox and @trustin on this decoration order API.
- As you said, this API is really useful for users. Maybe no.
- This can increase the complexity of decoration API and cause another confusion on the decoration order.

@heowc Sorry for my misdirection. 🙇‍♂️ I think our decoration order problem was fixed by #3166. Don't you mind closing this PR?

Reply:
Fine. 😊 I will close the PR and back up the branch.

Reply:
I am looking forward to your other awesome PRs. 😉

Reply:
Sorry for making you wasted time, @heowc. 🙇